js/interface.js

```diff
@@ -315,14 +313,100 @@
     });
   }
 
-  function loadCustomMenuWidgets() {
-    $('.menu-styles-wrapper').addClass('loading');
+  function isNewerVersion(latestVersionMenu, currentMenuVersion) {
+    for (var index = 0; index < latestVersionMenu.length; index++) {
+      if (isNaN(+latestVersionMenu[index])) latestVersionMenu[index] = 0;
+
+      if (isNaN(+currentMenuVersion[index])) currentMenuVersion[index] = 0;
+
+      if (+latestVersionMenu[index] > +currentMenuVersion[index]) return true;
+
+      if (+latestVersionMenu[index] < +currentMenuVersion[index]) return false;
+    }
+
+    return false;
+  }
+
+  /**
+   * Function to filter organizationMenus by version.
+   * @param {Array} data - array of organizationMenus
+   * @returns {Object} latestVersionMenu - object which contains the latest version of current menu package type
+   */
+
+  function getLatestMenuVersion(data) {
+    var latestVersionMenu;
+
+    data.forEach(function(menu) {
+      var formattedMenuVersion = menu.version.split('.');
+      var formattedLatestVersionMenu = latestVersionMenu ? latestVersionMenu.version.split('.') : null;
+
+      if (latestVersionMenu) {
+        latestVersionMenu = isNewerVersion(formattedLatestVersionMenu, formattedMenuVersion) ? latestVersionMenu : menu;
+      }
+
+      if (!latestVersionMenu) {
+        latestVersionMenu = menu;
+      }
+    });
+
+    return latestVersionMenu;
+  }
+
+  /**
+   * Creates a list of menu widgets to be displayed based on versions and current usage
+   * @param {Array} menus - List of menu widgets
+   * @returns {Array} List of menu widgets to be displayed based on versions and current usage
+   */
 
+  function generateMenuList(menus) {
+    var menusByPackage = _.groupBy(menus, 'package');
+    var menuList = [];
+
+    _.forIn(menusByPackage, function(menuVersions) {
+      var currentVersion = _.find(menuVersions, function(menuVersions) {
+        return menuVersions.instances.length;
+      });
+
+      // A version is currently in use
+      if (currentVersion) {
+        return menuList.push(currentVersion);
+      }
+
+      // Find latest version of menu
+      menuList.push(getLatestMenuVersion(menuVersions));
+    });
+
+    // Sort displayed menus by display name
+    return _.sortBy(menuList, 'name');
+  }
+
+  function loadCustomMenuWidgets() {
     return fetchCustomMenuWidgets().then(function(menus) {
-      customMenus = menus;
+      var menusWithInstances = _.filter(menus, function(menu) {
+        return menu.instances.length;
+      });
+
+      // If there is more than 1 menu with instances, clean up
+      if (menusWithInstances.legnth > 1) {
+        // Keep the first one found
+        menusWithInstances.shift();
```

Comment:
What if the first menu with instances has multiple instances? The scenario isn't considered in your code.

Reply:
@tonytlwu, Can you please suggest how to reproduce this scenario and what should be expected from it?

Reply:
If you comment out the code that deletes the instances, then you can add as many menu instances as you'd like to the app. We're looking for this scenario:

1. Menu 1: has 2 instances (A and B)
2. Menu 2: no instances
3. Menu 3: has 2 instances (C and D)
4. Menu 4: has 1 instance (E)
5. Menu 5: no instances

Only Instance A should remain. All others should be deleted.

Reply:
@tonytlwu, i hope i understand your comment above correctly.
https://github.com/Fliplet/fliplet-widget-menu-manager/pull/40/commits/e8c4c47c192a4e289d09907f3235502002dfe67e#diff-3e00189c2b38ee2b81964659c3a01eceef437f47aeefe02a20fa6f8d876a938fR394

Reply:
@AndrRyaz Have you tested your code to confirm it works?

Reply:
@tonytlwu, now tested, sorry, I didn’t do the `fliplet publish last time and thought that everything was working correctly, this will not happen anymore.

I have a question if instanceToDelete can contain more than 1 ID, because I tried to reproduce this but i cant

Reply:
@squallstar Can you check whether the endpoint we're using here can delete multiple instances in one go? It's rare that an app would have so many menus to delete that it would create a significant load, which means we could use `Promise.all()` to call the endpoint multiple times at once. What's your judgement on this?

@AndrRyaz The logic can be simplified much further.

1. Take `menus` and grab all the instance IDs.
2. If the number existing instance ID is less than 2, do nothing
3. Ignore the first instance ID and delete all the rest

Reply:
@tonytlwu we currently only have an endpoint to delete one instance at time (ref https://github.com/Fliplet/fliplet-api/blob/master/routes/v1/widget-instances.js#L650)

However, it’s fairly easy to add a new endpoint to delete multiple at once.

Reply:
@squallstar How much time do you need?

Reply:
@tonytlwu 2 hours

Reply:
created a task https://weboo.atlassian.net/browse/ID-622

Reply:
@squallstar hi, can you please give me the endpoint to delete multiple id's please

Reply:
As mentioned to @sofiiakvasnevska, there is no endpoint to remove an array of IDs. Please make one request per ID (using 
 `Promise.all` if applicable) to the endpoint.

Reply:
@tonytlwu hi, you can please check if I understand your proposal correctly. I need to collect all records that have an instance and make a check that if the length of the array is more than 1, then delete all records in this array except the first one, but I have a question if my bottom bar type will have an array of 1 instance and for the menu type kirkle there will be an array of 2 inputs, it turns out that there will be two menus with instances

Reply:
@tonytlwu i think i understand your comment, this is not relevant anymore